pkgs/build-support/fetchpatch/default.nix

```diff
@@ -12,8 +12,26 @@ in
 { stripLen ? 0, extraPrefix ? null, excludes ? [], includes ? [], revert ? false, ... }@args:
 
 let
+  urls_ =
+    if args ? urls && args.urls != [] && !(args ? url) then
+      (if lib.isList args.urls then args.urls
+       else throw "`urls` is not a list")
+    else if args ? url && args.url != "" && !(args ? urls) then
+      (if lib.isString args.url then [args.url]
+       else throw "`url` is not a string")
+    else throw "fetchpatch requires either `url` or `urls` to be set";
+
+  hash =
+    if args ? hash then { outputHashAlgo = null; outputHash = args.hash; }
+    else if args ? md5 then throw "fetchpatch does not support md5 anymore, please use sha256 or sha512"
+    else if (args ? outputHash && args ? outputHashAlgo) then { inherit (args) outputHashAlgo outputHash; }
+    else if args ? sha512 then { outputHashAlgo = "sha512"; outputHash = args.sha512; }
+    else if args ? sha256 then { outputHashAlgo = "sha256"; outputHash = args.sha256; }
+    else if args ? sha1 then { outputHashAlgo = "sha1";   outputHash = args.sha1; }
```

Comment:
sha1 wasn't supported before and I am not sure if we should add support for weaker hash algorithms. I think we should treat this the same way as md5.